pinot-common/src/main/java/org/apache/pinot/common/lineage/SegmentLineageUtils.java

```diff
@@ -53,4 +54,19 @@ public static void filterSegmentsBasedOnLineageInPlace(Set<String> segments, Seg
       }
     }
   }
+
+  public static Set<String> getReplacedSegments(SegmentLineage segmentLineage) {
```

Comment:
can we modify the `filterSegmentsBasedOnLineageInPlace` to use `getReplacedSegments`? In that way, we can keep the core logic in a single place.

```
Set<String> replacedSegments = getReplacedSegments()
segments.removeAll(replacedSegments)
```